uisdk/src/test/java/tech/dojo/pay/uisdk/data/paymentintent/RefreshPaymentIntentRepositoryTest.kt

```diff
@@ -0,0 +1,106 @@
+package tech.dojo.pay.uisdk.data.paymentintent
+
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doAnswer
+import org.mockito.kotlin.given
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import tech.dojo.pay.sdk.card.entities.CardsSchemes
+import tech.dojo.pay.uisdk.data.entities.PaymentIntentPayload
+import tech.dojo.pay.uisdk.data.mapper.PaymentIntentPayLoadMapper
+import tech.dojo.pay.uisdk.domain.entities.AmountDomainEntity
+import tech.dojo.pay.uisdk.domain.entities.PaymentIntentDomainEntity
+import tech.dojo.pay.uisdk.domain.entities.RefreshPaymentIntentResult
+import tech.dojo.pay.uisdk.domain.mapper.PaymentIntentDomainEntityMapper
+
+class RefreshPaymentIntentRepositoryTest {
+    private val dataSource: PaymentIntentDataSource = mock()
+    private val paymentIntentDomainEntityMapper: PaymentIntentDomainEntityMapper = mock()
+    private val paymentIntentPayLoadMapper: PaymentIntentPayLoadMapper = mock()
+    private lateinit var sut: RefreshPaymentIntentRepository
+
+    @Before
+    fun setUp() {
+        sut = RefreshPaymentIntentRepository(dataSource, paymentIntentDomainEntityMapper, paymentIntentPayLoadMapper)
+    }
+
+    @Test
+    fun `when refreshPaymentIntent fails refresh PaymentIntent stream should emit RefreshFailure`() = runTest {
+        // arrange
+        val paymentId = "paymentId"
+        doAnswer {
+            val onFailure: () -> Unit = it.arguments[2] as () -> Unit
+            onFailure.invoke()
+        }.whenever(dataSource).refreshPaymentIntent(any(), any(), any())
```

Comment:
```suggestion
given(dataSource. refreshPaymentIntent(any(), any(), any()))
      .willAnswer {
          val onFailure = it.arguments[2] as () -> Unit
          onFailure.invoke()
      }
```

The same for other tests in this file